luigi/scheduler.py

```diff
@@ -655,9 +672,15 @@ def add_task(self, task_id=None, status=PENDING, runnable=True,
         if deps is not None:
             task.deps = set(deps)
 
+        if deps_configs is not None:
+            task.deps_configs = deps_configs
+
         if new_deps is not None:
             task.deps.update(new_deps)
 
+        if new_deps_configs is not None:
+            task.deps_configs.update(new_deps_configs)
```

Comment:
On scheduler side, whenever a task is tried to be added, its retry-policy should also be sent from wherever it is invoked. If it is not sent, the config in luigi.cfg will be active for retry-policy. To provide that, I looked all places which invoke scheduler `_add_task` method and also made them send task-level config. I saw two kind of places which  are runing task normally and handling dynamic dependencies. I though that, even a task is added as dynamically, its task-level config must be passed to add_task method to apply task-level config of those kind of tasks. Am I thinking wrongly?